producer-kafka/README.adoc

```diff
@@ -0,0 +1,407 @@
+      },
+      "col_1":{
+        "value": null
+      }
+   }
+}
+----
+Please note that the `col_1` contains the `value`, but the actual value is equal to `null`.
+
+==== Change of the value
+
+Finally, the most obvious state is denoting that there was a change to the particular field.
+The resulting CDC event will contain the value with a change:
+[source,json]
+----
+{
+   "op":"d",
+   "ts_ms":0,
+   "data":{
+      "pk_1":{
+         "value":"pk_value"
+      },
+      "ck_1":{
+         "value":1
+      },
+      "col_1":{
+         "value": "new_value"
+      }
+   }
+}
+----
+The `new_value` means a change of the `col_1` cell in the underlying database.
+
+The value schema is more flexible regarding keeping the compatibility.
+This flexibility is achieved by using a union type that allows the evolution of schema in a backward-compatible way.
+It will enable renaming, adding, and removing of Clustering Key and Cells.
+To make it work, you need to assure that the consumer will read the Avro record using proper schema id.
+
+
+=== Schema Supported Types
+The resulting schema supports a variety of types that can be used for any cell value.
+
+==== Supported Native Types.
+Let's take a look at a list of supported native types:
+
+- `Native.ASCII` is serialized as avro `Type.STRING`.
+- `Native.BIGINT` is serialized as avro `Type.LONG`.
+- `Native.BLOB` is serialized as avro `Type.BYTES`.
+- `Native.BLOB` is serialized as avro `Type.BYTES`.
+- `Native.BOOLEAN` is serialized as avro `Type.BOOLEAN`.
+- `Native.COUNTER` is serialized as avro `Type.LONG`.
+- `Native.DATE` is serialized as avro logical `date` type with underlying `Type.INT` representation.
+- `Native.DECIMAL` is serialized as avro logical `decimal` type with underlying `Type.BYTES` representation. It uses `10` as a default decimal precision.
+- `Native.DOUBLE` is serialized as avro `Type.DOUBLE`.
+- `Native.DURATION` is serialized as a raw `Type.BYTES`
+- `Native.FLOAT` is serialized as avro `Type.FLOAT`.
+- `Native.INET` is serialized as a raw `Type.BYTES`
+- `Native.INT` is serialized as avro `Type.INT`.
+- `Native.SMALLINT` is serialized as avro logical `short` type with underlying `Type.INT` representation.
+- `Native.TEXT` is serialized as avro `Type.STRING`.
+- `Native.TIME` is serialized as avro logical `timeMicros` type with underlying `Type.LONG` representation.
+- `Native.TIMESTAMP` is serialized as avro logical `timestampMillis` type with underlying `Type.LONG` representation.
+- `Native.TIMEUUID` is serialized as avro logical `uuid` type with underlying `Type.STRING` representation.
+- `Native.TINYINT` is serialized as avro logical `byte` type with underlying `Type.INT` representation.
+- `Native.UUID` is serialized as avro logical `uuid` type with underlying `Type.STRING` representation.
+- `Native.VARCHAR` is serialized as avro `Type.STRING`.
+- `Native.TIMESTAMP` is serialized as avro logical `bigInteger` type with underlying `Type.BYTES` representation.
```

Comment:
Is this a typo? There's this one up above

```
- `Native.TIMESTAMP` is serialized as avro logical `timestampMillis` type with underlying `Type.LONG` representation.
```

Reply:
this was a typo, fixed now